### ISSUE TITLE
fix Seq.of flow error for immutable@^3

### ISSUE THREAD

**Issue by jedwards1211** [facebook/immutable-js#1435]
After upgrading to Flow 0.56, it looks like, for better or worse, Flow is going to require static methods on a derived class to return a subtype of what the same static method on the base class returns.  This causes the following error:

```
Error: node_modules/immutable/dist/immutable.js.flow:399
399:   static of<T>(...values: T[]): SetSeq<T>;
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ function type. This type is incompatible with
380:   static of<T>(...values: T[]): IndexedSeq<T>;
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ function type
  This parameter is incompatible:
    399:   static of<T>(...values: T[]): SetSeq<T>;
                                         ^^^^^^^^^ SetSeq. This type is incompatible with
    380:   static of<T>(...values: T[]): IndexedSeq<T>;
                                         ^^^^^^^^^^^^^ IndexedSeq


Found 1 error
```

I wanted to submit a PR to fix this but I can't but there's no branch for version 3.x.x.  This error can be fixed by changing line 380 from
```js
static of<T>(...values: T[]): IndexedSeq<T>;
```
to
```js
static of<T>(...values: T[]): Seq<any, T>;
```

Could you make that change and release it?

Someone may be depending on the fact that `Seq.of` returns an `IndexedSeq`...but they could fix that by using `IndexedSeq.of` instead.

**Comment by tanem** [comment]
@jedwards1211 this was discussed & patched on the v4 branch in https://github.com/facebook/immutable-js/issues/1308 if that's helpful (we're still using v3 as well so in the same boat).

**Comment by ksmithbaylor** [comment]
Any update on this? It's currently the only Flow failure in my codebase and I'd love to see it gone :)

**Comment by TomiS** [comment]
Is this dupe of https://github.com/facebook/immutable-js/issues/1308 ?

**Comment by zGrav** [comment]
Still waiting on this :(

**Comment by jdeniau** [comment]
@zGrav @TomiS @ksmithbaylor
I'm interested to know if you actually still use flow in your codebase after all this years?
Immutable flow support is quite inexistant, and the community seems to have migrate to Typescript, so I'm thinking of removing flow support in immutable to me honest.
I'd like your insight here!
Thanks

**Comment by TomiS** [comment]
Oh my, how the years go by... Thanks for the concern @jdeniau , but I agree it's probably time to move on. We too have migrated off from flow years ago already (to ReScript, in fact). But either way, I think everyone have had plenty of time to adapt at this point. But that's ofc only my 2c.

**Comment by ksmithbaylor** [comment]
Definitely, that was a few jobs ago and I'd be surprised if they still use flow.
